service/history/timerQueueProcessorBase.go

```diff
@@ -352,6 +352,7 @@ func (t *timerQueueProcessorBase) readAndFanoutTimerTasks() (*persistence.TimerT
 	t.lastPollTime = time.Now()
 	timerTasks, lookAheadTask, moreTasks, err := t.timerQueueAckMgr.readTimerTasks()
 	if err != nil {
+		t.notifyNewTimer(time.Time{}) // re-enqueue the event
```

Comment:
Can we end up in a tight loop here?  
Probably it is safer to do a backoff here.  Or do we have backoff at the persistence level?

Reply:
just few line above, there is a rate limiter, which will protect the DB